### DISCUSSION TITLE
Is there a way of making the notifications from WALC to show in my notification history (in KDE Plasma)?

### DISCUSSION THREAD

**Discussion by albakhtari** [cstayyab/WALC#176] [Feature Requests]
> Is there a way of making the notifications in from WALC to show in my notification history (in KDE Plasma)?
Currently, it displays the notification for a few seconds then disappears.

Can you force electron to use `notify-send`?

![Screenshot_20220417_152020](https://user-images.githubusercontent.com/79008923/163710387-4523c1f6-375e-4a6f-82af-0a23e207579c.png)

![Screenshot_20220417_152114](https://user-images.githubusercontent.com/79008923/163710413-0faa1e72-22b1-43e7-9bfb-f2f67fd3d7d0.png)

I have no experience in electron, although I'm quite good at JS, but if there is a way to get electron to use `notify-send` then I'd like to contribute to the project and do it myself.

_Originally posted by @YoungFellow-le in https://github.com/WAClient/WALC/discussions/174#discussioncomment-2581285_

**Comment by albakhtari** [comment]
```js
	function doNotify(){
			Notification.requestPermission().then(function (result){
				var myNotification = new Notification('The guy that sent me the message...', {
					'body': 'Test Notification',
					'icon': 'https://upload.wikimedia.org/wikipedia/commons/thumb/6/6b/WhatsApp.svg/1200px-WhatsApp.svg.png'
				});
			});
		}
```
This worked :point_up: 
However, this isn't the solution, you can't access this part of the code...

**Comment by daffamumtaz2361** [comment]
Yeah, not really sure why it does that, also sometimes the notification just flashes in and out quickly. I'll see what I can do. It is a bit tricky since I can't change WhatsApp's code

**Reply by albakhtari** [reply]
Maybe you can change the way electron translates the HTML5 notifications?

**Comment by daffamumtaz2361** [comment]
So here's what I found. WALC is not registered as application by KDE, it's registered as "Other Applications". So if you uncheck the "Show in history" option for "Other Applications", WALC notification won't appear in history. I tried using `notify-send` but it also registered as Other Applications so it has the same problem.

The question here is how to make WALC registered as it's own application by KDE. I tried messing with desktop file but so far no luck.

**Reply by albakhtari** [reply]
After running this command:
```shell
 notify-send --app-name=WALC --hint=string:desktop-entry:WALC -i dialog-information -t 10000 "Can we have notifications like this?"
```
WALC is added to the applications' notification settings:

![Screenshot_20220419_134956](https://user-images.githubusercontent.com/79008923/163964834-055bc543-973e-4c17-aba2-3f45cf3c1827.png)

However, this didn't help.

Then I checked these boxes :point_down:

![Screenshot_20220419_135127](https://user-images.githubusercontent.com/79008923/163965829-4307dea1-b9ae-4730-967e-4e9f908e0c94.png)

 And it worked.... But like this:

![Screenshot_20220419_135346](https://user-images.githubusercontent.com/79008923/163967633-28853f76-4332-4a81-9409-2916ac01641d.png)

Which is fine I guess....

**Comment by daffamumtaz2361** [comment]
After a discussion with a friendly and helpful KDE developer whom also implemented the cool notification inline reply, this will be fixed along with adding inline reply!

However, implementing this is quite complicated since I have to rewrite the notification from scratch, so that will come after 0.2.2

**Reply by albakhtari** [reply]
That's great!
Just make sure it doesn't take you away from your عبادة in رمضان, we can live without inline replies...but this رمضان may be our last....

**Reply by daffamumtaz2361** [reply]
Yes of course, I'm doing this in my free time after all
